From d42f21a0b52f5b7bf8d4de1c58d21f23bd28f64a Mon Sep 17 00:00:00 2001
From: Vladimir Kostyukov <vladimir.kostyukov@intel.com>
Date: Mon, 2 Dec 2013 13:29:53 +0700
Subject: Dalvik: PCG does not work in host-mode

BZ: 155444

There should be a dynamic-link dependency from libdvmpcgjit.so
to libdvm.so in host mode. This patch adds missed declaration
into the PCG's Android.mk makefile.

Change-Id: I48f618277fd703385ea36c67b225fc325590acd7
Category: device enablement
Domain: AOSP.Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: Vladimir Kostyukov <vladimir.kostyukov@intel.com>
Signed-off-by: Vitaly Provodin <vitaly.a.provodin@intel.com>
---
 vm/compiler/codegen/x86/pcg/Android.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/Android.mk b/vm/compiler/codegen/x86/pcg/Android.mk
index 0c426fd..02cb79d 100644
--- a/vm/compiler/codegen/x86/pcg/Android.mk
+++ b/vm/compiler/codegen/x86/pcg/Android.mk
@@ -166,6 +166,7 @@ ifeq ($(WITH_HOST_DALVIK),true)
     LOCAL_ALLOW_UNDEFINED_SYMBOLS := true
     TARGET_LIBGCC = $(shell $(TARGET_CC) -m32 -print-libgcc-file-name)
     LOCAL_STATIC_LIBRARIES += libpcg_host libirc_pcg libsvml_pcg
+    LOCAL_SHARED_LIBRARIES += libdvm
     LOCAL_SRC_FILES := $(PCG_SRC_FILES)
     LOCAL_C_INCLUDES += $(PCG_C_INCLUDES)
 
-- 
1.7.4.1

